Refactor: TypeScript Migration and Project Modernization

dimensi opened the pull request:
## Summary

This MR includes a major refactoring of the hotkeys-js project: migration to TypeScript, dependency updates, build system modernization, and layout independence improvements.

## Key Changes

- **TypeScript migration**: Full migration from JavaScript to TypeScript with proper type definitions
- **Layout independence**: Added keyboard layout-independent hotkey detection using `event.code` mapping (e.g., `ctrl+a` now works consistently on Russian, German, and other keyboard layouts)
- **Build system**: Migrated from legacy build tools to Vite
- **ESLint**: Updated to v8.57.1 with new flat config (`eslint.config.js`), removed deprecated config files
- **Dependencies**: Updated all packages to latest versions
- **Styling**: Migrated from LESS to CSS modules

## Note

> Using AI for writing code isn't ideal, but at least it's done. Hope the result is decent despite using AI tools during development.

jaywcjlove commented on dist/hotkeys.js:
@dimensi The JavaScript files in the `dist` folder must not be removed, and [UNPKG](https://unpkg.com/hotkeys-js/dist/) can provide CDN access for the .js files in that directory.

```html
<script src="https://unpkg.com/hotkeys-js/dist/hotkeys.min.js"></script>
```

jaywcjlove commented:
<img  src="https://github.com/user-attachments/assets/5c8f5d66-ba1c-49a6-8d65-562b99c1be60" />
Thanks! But the test cases didn’t pass. 
@dimensi

dimensi commented:
@jaywcjlove oops, my mistake — while moving files back and forth, I accidentally broke the tests.

Fixed the tests and corrected the file path.
Updated the links in the README — realized they were pointing to old URLs.
Ended up bringing back the dist folder, since I didn’t notice you were also using Statically and Githack as link sources. These services pull files directly from Git, so I had to restore it

jaywcjlove commented:
@dimensi 

https://github.com/jaywcjlove/hotkeys-js/blob/04f097fdae92e27fecd1a595326274a397192edb/README.md?plain=1#L471-L473
